[CI] SpanGapQueryBuilderTests testToQuery failing

**Build Scans:**
- [elasticsearch-intake #10727 / part1](https://gradle-enterprise.elastic.co/s/tx75fqzkzpp34)
- [elasticsearch-pull-request #34814 / part-1](https://gradle-enterprise.elastic.co/s/shvnqtoapzqp4)
- [elasticsearch-pull-request #34813 / part-1](https://gradle-enterprise.elastic.co/s/px73uztis4kku)
- [elasticsearch-intake #10726 / part1](https://gradle-enterprise.elastic.co/s/xcuxmwgwuxoos)
- [elasticsearch-pull-request #34659 / part-1](https://gradle-enterprise.elastic.co/s/ut6oghdrobj6k)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.query.SpanGapQueryBuilderTests.testToQuery" -Dtests.seed=7A5C666985480608 -Dtests.locale=ja -Dtests.timezone=America/St_Kitts -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testToQuery'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.index.query.SpanGapQueryBuilderTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: ((an instance of org.apache.lucene.queries.spans.SpanNearQuery or an instance of org.apache.lucene.queries.spans.SpanTermQuery) or an instance of org.elasticsearch.index.query.MatchAllQueryBuilder)
     but: was <SpanMatchNoDocsQuery("unmapped field: xL")>
```

**Issue Reasons:**
- [main] 2 consecutive failures in step part1
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 5 failures in test testToQuery (1.6% fail rate in 316 executions)
- [main] 2 failures in step part1 (8.3% fail rate in 24 executions)
- [main] 3 failures in step part-1 (4.2% fail rate in 72 executions)
- [main] 2 failures in pipeline elasticsearch-intake (8.3% fail rate in 24 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (4.3% fail rate in 70 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -371,9 +371,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/114187
 - class: org.elasticsearch.xpack.esql.action.EsqlActionBreakerIT
   issue: https://github.com/elastic/elasticsearch/issues/114194
-- class: org.elasticsearch.index.query.SpanGapQueryBuilderTests
-  method: testToQuery
-  issue: https://github.com/elastic/elasticsearch/issues/114218
 
 # Examples:
 #

--- a/server/src/test/java/org/elasticsearch/index/query/SpanGapQueryBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/query/SpanGapQueryBuilderTests.java
@@ -13,6 +13,7 @@ import org.apache.lucene.queries.spans.SpanNearQuery;
 import org.apache.lucene.queries.spans.SpanQuery;
 import org.apache.lucene.queries.spans.SpanTermQuery;
 import org.apache.lucene.search.Query;
+import org.elasticsearch.lucene.queries.SpanMatchNoDocsQuery;
 import org.elasticsearch.test.AbstractQueryTestCase;
 
 import java.io.IOException;
@@ -50,7 +51,9 @@ public class SpanGapQueryBuilderTests extends AbstractQueryTestCase<SpanNearQuer
     protected void doAssertLuceneQuery(SpanNearQueryBuilder queryBuilder, Query query, SearchExecutionContext context) throws IOException {
         assertThat(
             query,
-            either(instanceOf(SpanNearQuery.class)).or(instanceOf(SpanTermQuery.class)).or(instanceOf(MatchAllQueryBuilder.class))
+            either(instanceOf(SpanNearQuery.class)).or(instanceOf(SpanTermQuery.class))
+                .or(instanceOf(MatchAllQueryBuilder.class))
+                .or(instanceOf(SpanMatchNoDocsQuery.class))
         );
         if (query instanceof SpanNearQuery spanNearQuery) {
             assertThat(spanNearQuery.getSlop(), equalTo(queryBuilder.slop()));

